ngraph/frontend/paddlepaddle/src/frontend.cpp

```diff
@@ -75,22 +75,64 @@ bool endsWith(const std::string &str, const std::string &suffix) {
     return false;
 }
 
+template<typename T>
+std::vector<T> reinterpret_vector(const std::vector<uint8_t>& a) {
+    PDPD_ASSERT(a.size() % sizeof(T) == 0);
+    std::vector<T> res;
+    res.reserve(a.size() / sizeof(T));
+    for (size_t i = 0; i < a.size(); i += sizeof(T)) {
+        res.push_back(*reinterpret_cast<const T*>(&a[i]));
+    }
+    return res;
+}
+
 } // namespace pdpd
 
-std::shared_ptr<Constant> FrontEndPDPD::read_tensor(const std::shared_ptr<TensorPlacePDPD>& tensor_place,
-                const std::shared_ptr<InputModelPDPD>& model)
+std::shared_ptr<Node> FrontEndPDPD::make_const_node(const std::shared_ptr<TensorPlacePDPD>& tensor_place,
+                                                    const std::shared_ptr<InputModelPDPD>& model)
 {
     const auto& var_desc = tensor_place->getDesc();
     std::cout << "Reading tensor " << var_desc->name() << std::endl;
     PDPD_ASSERT(var_desc->type().type() == paddle::framework::proto::VarType::LOD_TENSOR);
     const auto& tensor = var_desc->type().lod_tensor().tensor();
     const auto& tensor_length = std::accumulate(
         tensor.dims().cbegin(), tensor.dims().cend(), 1, std::multiplies<int64_t>());
-    // TODO: implement for other types
-    auto tensor_data = model->readWeight(var_desc->name(), tensor_length);    
-
-    std::vector<size_t> shape(tensor.dims().cbegin(), tensor.dims().cend());
-    return Constant::create(element::f32, Shape(shape), tensor_data);
+    Shape shape(tensor.dims().cbegin(), tensor.dims().cend());
+    const auto& type = TYPE_MAP[tensor.data_type()];
+    switch (type) {
+        case element::f32:
```

Comment:
for `type `we have `.size()` method, it returns bytesize of element type
`openvino/ngraph/core/src/type/element_type.cpp` line 123

I think we can simplify all readWeights calls, or even replace them with one call 
`auto tensor_data = model->readWeight(var_desc->name(), tensor_length * type.size());`